Adding graphs to base DAShboard

rileyclarke opened the pull request:
Added three of the graphs. Had to create new JS file to hold code. has a function that should input the new data into the graphs, needs to be tested still. Will also need to make a listening function to resize the graphs when the window is resized, at the moment it needs to be refreshed when the window size is changed.

rileyclarke committed:
Adding graphs to base DAShboard

Added three of the graphs. Had to create new JS file to hold code. has a function that should input the new data into the graphs, needs to be tested still. Will also need to make a listening function to resize the graphs when the window is resized, at the moment it needs to be refreshed when the window size is changed.

khanguslee reviewed (changes requested):
Looking good! Just some small changes to make. 

- Rename main.js to index.js. Good convention to keep the js file the same name as the html file. (If there's only one) To do this properly, **don't** rename the file like you would normally do. In the command line, type `git mv main.js index.js`.

- Add axis titles

khanguslee commented on client/index.html:
```diff
@@ -27,9 +27,13 @@
-				<h1>Dashboard</h1>
-			</div>
+
+            <div id="chart1" style="width: 45%; height: 45vh; float: left;"><canvas id="velocity-time"></canvas></div>
```
Can you rename all the canvas ids to `velocityTimeChart`, `powerTimeChart` etc

khanguslee replied:
Rename id of the divs `chart1`, `chart2`, etc. to something that's a bit more verbose eg. `velocityTimeChartDiv`

khanguslee commented on client/js/main.js:
```diff
@@ -0,0 +1,120 @@
+document.getElementById("velocity-time").width = document.getElementById("chart1").offsetWidth;
```
Instead of writing `document.getElementById("velocity-time")` everywhere, you can set it to a variable instead!

```
let velocityTimeChart = document.getElementById("velocity-time");
let velocityChartDiv= document.getElementById("chart1");

velocityTimeChart.width = velocityChartDiv.offsetWidth;
```

khanguslee commented on client/js/main.js:
```diff
@@ -0,0 +1,120 @@
+                    }]
+                    },
+
+                options:{
```
I would recommend creating a variable called `velocityTimeChartOptions` and storing what is here instead. (Do for the other charts too)

khanguslee commented on client/index.html:
```diff
@@ -3,9 +3,9 @@
 	<head>
 		<script src="js/jquery.min.js"></script>
 		<script src="js/bootstrap.bundle.min.js"></script>
+        <script src="https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.1.4/Chart.min.js"></script>      
```
This works when our computers have internet. But won't work when the DAS is in the countryside of Victoria 😢. I'm happy to fix this for you though. Just letting you know!

khanguslee reviewed (changes requested):
Let me know if you need help with any of these!

rileyclarke committed:
Making changes to graphs

Implemented all recommended changes, apart from file name change.

html element names have been changed, variable created for document elements and chart options to simplify js file. Axis labels have been added and the graph layout has been changed (graphs on right are set to align to the right).

khanguslee commented on client/js/main.js:
```diff
@@ -0,0 +1,161 @@
+              };
+
+
+new Chart(document.getElementById("velocityTimeChart"), {
```
Instead of the document.getElementById("..."), can you put in the variable that you set up top?

khanguslee reviewed (changes requested):
Only change that is needed!

rileyclarke committed:
Add use of variable name

Changed chart assignment to use previously declared variable

khanguslee committed:
Rename file to index.js

khanguslee committed:
Merge branch 'graph-display' of https://github.com/Monash-Human-Power/MHP-DAS-Web-Server into graph-display

khanguslee committed:
Change script reference of main.js to index.js

khanguslee committed:
Install chart.js via npm and use grunt to include the package

khanguslee commented:
- Renamed main.js file to index.js
- Removed the chart.js cdn and installed chart.js via npm and then used gulp.
